src/openrct2/scripting/bindings/entity/ScPeep.hpp

```diff
@@ -147,10 +149,12 @@ namespace OpenRCT2::Scripting
         {
             ThrowIfGameStateNotMutable();
             auto peep = GetPeep();
-            if (peep != nullptr)
-            {
-                peep->Energy = value;
-            }
+            if (peep == nullptr)
+                return;
+            if (value < kPeepMinEnergy || value > kPeepMaxEnergy)
```

Comment:
Maybe it's better to clamp the value than to have nothing happen when setting the energy? 

Also, what about peeps that have an energy between 1 to `kPeepMinEnergy`? (I don't know if they exist)

Reply:
> Maybe it's better to clamp the value than to have nothing happen when setting the energy?

Agreed, that might be better than returning outright.

> Also, what about peeps that have an energy between 1 to kPeepMinEnergy? (I don't know if they exist)

Better to be safe and reset them to `kPeepMinEnergy` as well, indeed, and not just the ones with 0 energy. Though I think we should only set the flags for the ones with 0 energy -- agreed?